### ISSUE TITLE
Endless loop of "building package" when running two separate meteor codebases on different ports that share a local package

### ISSUE THREAD

**Issue by nadeemja** [meteor/meteor#8656]
Hi,

I have two codebases with meteor 1.4.4.1 on El Capitan. 

I'm trying to move a certain set of features from one to the other via a package. So I'm moving all the relevant code into the package, and making sure that the originator codebase still works. In the receiving codebase I'm only doing "meteor add" to install the package. 

I need to keep the package private, so pushing to atmosphere is a no go. Therefore, I've got the package in a .devPackages/MyPackage folder and in packages/ folder I'm symlinking to .devPackages. In the recipient repo I'm symlinking from packages/myPackage to the .devPackages folder in the originator repo. This works fine, except for when I run the two codebases simultaneously on different ports. 

The first time I got this error:

      /Users/Nadeem/.meteor/packages/standard-minifier-js/.2.0.0.1kun4p5++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:190

          throw error;
          ^
    Error: ENFILE: file table overflow, uv_cwd
        at Error (native)
        at setopts (/Users/Nadeem/.meteor/packages/meteor-tool/.1.4.4_1.3dyhv9++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/rimraf/node_modules/glob/common.js:91:21)
        at new Glob (/Users/Nadeem/.meteor/packages/meteor-tool/.1.4.4_1.3dyhv9++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/rimraf/node_modules/glob/glob.js:117:3)
        at Function.glob.hasMagic (/Users/Nadeem/.meteor/packages/meteor-tool/.1.4.4_1.3dyhv9++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/rimraf/node_modules/glob/glob.js:87:11)
        at Function.rimrafSync [as sync] (/Users/Nadeem/.meteor/packages/meteor-tool/.1.4.4_1.3dyhv9++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/rimraf/rimraf.js:266:36)
        at Object.rm_recursive (/tools/fs/files.js:303:12)
        at Builder.abort (/tools/isobuild/builder.js:696:11)
        at /tools/isobuild/bundler.js:2548:13
        at /tools/isobuild/bundler.js:2837:22
        at /tools/utils/buildmessage.js:271:13
        at [object Object]._.extend.withValue (/tools/utils/fiber-helpers.js:89:14)
        at /tools/utils/buildmessage.js:264:29
        at [object Object]._.extend.withValue (/tools/utils/fiber-helpers.js:89:14)
        at /tools/utils/buildmessage.js:262:18
        at [object Object]._.extend.withValue (/tools/utils/fiber-helpers.js:89:14)
        at /tools/utils/buildmessage.js:253:23
        at [object Object]._.extend.withValue (/tools/utils/fiber-helpers.js:89:14)
        at Object.capture (/tools/utils/buildmessage.js:252:19)
        at bundle (/tools/isobuild/bundler.js:2670:31)
        at /tools/isobuild/bundler.js:2617:32
        at Object.withCache (/tools/fs/files.js:1652:12)
        at Object.exports.bundle (/tools/isobuild/bundler.js:2617:16)
        at /tools/runners/run-app.js:588:36
        at Function.run (/tools/tool-env/profile.js:490:12)
        at bundleApp (/tools/runners/run-app.js:587:34)
        at [object Object]._.extend._runOnce (/tools/runners/run-app.js:631:35)
        at [object Object]._.extend._fiber (/tools/runners/run-app.js:890:28)
        at /tools/runners/run-app.js:417:12

To fix this error message I deleted node_modules in my myPackage folder and ran yarn.

Now, however, I'm getting and endless loop of this in both server instances in tandem. Once one instance is finished building the package, the other starts building, and once that one is done, the first one starts all over again, and round we go.

> Started proxy.
        sembly:studietid: updating npm dependencies -- nouislider, moment-range, tablesort, jquery...
        => Started MongoDB.
        => Linted your app. No linting errors.
        sembly:studietid: updating npm dependencies -- nouislider, moment-range, tablesort, jquery...
        => Linted your app. No linting errors.
        sembly:studietid: updating npm dependencies -- nouislider, moment-range, tablesort, jquery...
        => Linted your app. No linting errors.
        sembly:studietid: updating npm dependencies -- nouislider, moment-range, tablesort, jquery...
        => Linted your app. No linting errors.
        sembly:studietid: updating npm dependencies -- nouislider, moment-range, tablesort, jquery...

I don't think this is supposed to happen, is it?

**Comment by hwillson** [comment]
Hi @nadeemja - is your `.devPackages/MyPackage` folder within your Meteor app? What happens if you move that folder outside of your Meteor app directory structure, then symlink to it from the `packages` directory of both applications?

**Comment by r4ndr4s4** [comment]
I attached a GIF which shows what I experience in my new project using Meteor 1.5.1. Everytime I add or remove a package while my app is running, I stuck in a loop. I have to terminate the process and start the app again as it is inifinite and cause high CPU usage. (OSX 10.12.5) It may be relevant.
![jul -29-2017 12-42-11](https://user-images.githubusercontent.com/13931659/28744201-0ad62658-745c-11e7-9fd1-7eaab6cfceba.gif)

**Comment by nadeemja** [comment]
@hwillson 

I haven't had a chance to work on the same problem in a while, but will do so in a few days. 

You are correct that the .devPackages was inside the originator repo. Will attempt your suggestion and let you know.

**Comment by hwillson** [comment]
We haven't received answers to all of the questions asked in https://github.com/meteor/meteor/issues/8656#issuecomment-298356979, so I'll close this issue for now.

@r4ndr4s4 If your problem is still happening with Meteor 1.6, please consider opening a new issue.

Thanks!
